pincer/client.py

```diff
@@ -562,8 +566,38 @@ async def start_shard(self, shard: int, num_shards: int):
             }
         )
 
+        self.shards[gateway.shard] = gateway
         create_task(gateway.start_loop())
 
+    def get_shard(
+        self,
+        guild_id: Optional[Snowflake] = None,
+        num_shards: Optional[int] = None
+    ) -> Gateway:
+        """
+        Returns the shard receiving events for a specified guild_id.
+
+        ``num_shards`` is inferred from the num_shards value for the first started
+        shard. If your shards do not all have the same ``num_shard`` value, you must
+        specify value to get the expected result.
+
+        Parameters
+        ----------
+        guild_id : Optional[~pincer.utils.snowflake.Snowflake]
+            The guild_id of the shard to look for
+        num_shards : Optional[int]
+            The number of shards.
+        """
+        if not guild_id:
+            return self.shards[0]
```

Comment:
On L594 there is `if not self.shards`. This would imply a chance at the shards array being empty would it not? If so, this could throw an `IndexError`.